sumpy/kernel.py

```diff
@@ -263,6 +273,18 @@ def get_derivative_coeff_dict_at_source(self, expr_dict):
         """
         return expr_dict
 
+    def get_derivative_coeff_dict_at_target(self, expr_dict):
```

Comment:
- Describe better what the multi-indices on input and output side.
- Generate the source/target docstrings from one source via `format`.
- Name: `apply_target_transformation_to_derivative_coeff_dict`
- Maybe make an explicit type that wraps the `expr_dict` as you suggested.